### PR TITLE
Refactored the load script.

### PR DESCRIPTION
**Pull request by GabrielNsiah**
Refactored the load script for ticket #6. Includes:

- More logging.
- More error handling.
- No more -1 as a return value.
- Various code refactored.
- Verification for `.env` values.

### CODE REVIEW

File: pipeline/load.py [hunk 1]
```diff
@@ -44,65 +55,38 @@ def get_connection() -> extensions.connection:
...
+        return result[0] if result else None
+    except Exception as e:
+        logging.error("Error querying %s: %s", table_name, str(e))
+        return None
 
 
 def get_id_from_release_info(release_name: str, release_date: datetime,
-                             artist_id: int, type_id: int, cursor: extensions.cursor):
+                             artist_id: int, type_id: int,
+                             cursor: extensions.cursor) -> Optional[int]:
```
Comment (bentrzcinski) [inline pipeline/load.py:87]: Last week we got this feedback:
"Your return types can be dict | None, you don’t need to import optional and do Optional[dict]"
I think the same applies here

Comment (bentrzcinski) [reply pipeline/load.py:87]: And all the other instances in different functions

File: pipeline/load.py [hunk 2]
```diff
@@ -257,34 +241,73 @@ def insert_release_genres(release_id: int, genre_id: int, genre_name: str,
             "Error adding '%s' to '%s' ID: %s", genre_name, release_name, release_id)
 
 
+def insert_sale_data(sale_price: float, sale_date: datetime,
+                     country_name: str, release_id: int, cursor) -> None:
+    """"
+    Inserts the sale data relating to a release into the database
```
Comment (bentrzcinski) [inline pipeline/load.py:247]:
```suggestion
    Inserts the sale data relating to a release into the database.
```

File: pipeline/load.py [hunk 3]
```diff
@@ -24,7 +24,18 @@ def config_log() -> None:
...
-def get_connection() -> extensions.connection:
+def validate_env_vars() -> None:
+    """
+    Raises an error if any environment variables are missing
+    """
+    required_vars = ["DB_HOST", "DB_PORT", "DB_USER", "DB_PASSWORD", "DB_NAME"]
+    missing_vars = [var for var in required_vars if var not in environ]
+    if missing_vars:
+        raise EnvironmentError(
+            f"Missing required environment variables: {missing_vars}")
```
Comment (bentrzcinski) [inline pipeline/load.py:35]: This is really nice

**Review by bentrzcinski — CHANGES_REQUESTED** [review]
Really nice. I especially like all the error handling.

**Comment by GabrielNsiah** [comment]
Implemented the above changes, and also:
- Changed the formatting in `transform.py`, so that all dates are in the format `YYYY-MM-DD` for consistency.

**Review by bentrzcinski — APPROVED** [review]
great stuff

**Review by emily-curtis — APPROVED** [review]
looks good!

File: pipeline/load.py [hunk 4]
```diff
@@ -44,65 +54,38 @@ def get_connection() -> extensions.connection:
...
-        logging.warning("Unable to retrieve cursor for connection")
-        return None
+    except Exception as e:
+        logging.error("Unable to retrieve cursor for connection: %s", str(e))
+        raise
 
 
-def get_id_from_country(country_name: str, cursor: extensions.cursor) -> int:
+def get_id_from_table(search_value: str, table_name: str,
+                      cursor: extensions.cursor) -> int | None:
```
Comment (ebradley12) [inline pipeline/load.py:69]: Really nice - makes the code more DRY :)

File: pipeline/load.py [hunk 5]
```diff
@@ -139,30 +123,30 @@ def insert_country(country_name: str, cursor: extensions.cursor) -> None:
...
+            cursor.execute(insert_query, (artist_name,))
             logging.info("Artist '%s' was added to the database.", artist_name)
     except psycopg2.Error:
         logging.error("Error inserting artist: '%s'", artist_name)
 
 
-def insert_genres(genre_name: str, cursor: extensions.cursor) -> None:
+def insert_genres(genre_name: str, cursor: extensions.cursor) -> int | None:
     """
     Inserts genres into the database they don't already exist.
```
Comment (ebradley12) [inline pipeline/load.py:151]: Can you add 'if'

File: pipeline/transform.py [hunk 6]
```diff
@@ -57,7 +57,7 @@ def convert_from_unix_to_datetime(unix: str) -> datetime:
         return "None"
 
 
-def convert_date_format(date_str: str) -> str:
+def convert_written_date_format(date_str: str) -> str:
     """
     Takes dates in the "DD B YYYY" format and
     converts them to 'DD-MM-YYYY'.
```
Comment (ebradley12) [inline pipeline/transform.py:63]: Update docstrings to YYYY-MM-DD

**Comment by GabrielNsiah** [comment]
Inserted the above changes.

**Pull request merged**